flutter-idea/src/io/flutter/analytics/FlutterAnalysisServerListener.java

```diff
@@ -342,51 +367,85 @@ public void serverStatus(AnalysisStatus analysisStatus, PubStatus pubStatus) {
       errorCount += extractCount(errorCounts, AnalysisErrorType.COMPILE_TIME_ERROR);
       errorCount += extractCount(errorCounts, AnalysisErrorType.SYNTACTIC_ERROR);
-      FlutterInitializer.getAnalytics().sendEventMetric(DAS_STATUS_EVENT_TYPE, ERRORS, errorCount); // test: serverStatus()
-      int warningCount = 0;
+      warningCount = 0;
       warningCount += extractCount(errorCounts, AnalysisErrorType.STATIC_TYPE_WARNING);
       warningCount += extractCount(errorCounts, AnalysisErrorType.STATIC_WARNING);
-      FlutterInitializer.getAnalytics().sendEventMetric(DAS_STATUS_EVENT_TYPE, WARNINGS, warningCount); // test: serverStatus()
-      int hintCount = extractCount(errorCounts, AnalysisErrorType.HINT);
-      FlutterInitializer.getAnalytics().sendEventMetric(DAS_STATUS_EVENT_TYPE, HINTS, hintCount); // test: serverStatus()
-      int lintCount = extractCount(errorCounts, AnalysisErrorType.LINT);
-      FlutterInitializer.getAnalytics().sendEventMetric(DAS_STATUS_EVENT_TYPE, LINTS, lintCount); // test: serverStatus()
+      hintCount = extractCount(errorCounts, AnalysisErrorType.HINT);
+      lintCount = extractCount(errorCounts, AnalysisErrorType.LINT);
+      if (IS_TESTING) {
+        errorCount = warningCount = hintCount = lintCount = 1;
+      }
+      maybeReportErrorCounts();
     }
   }
 
+  private void maybeReport(boolean observeThrottling, @NotNull java.util.function.Consumer<@NotNull Analytics> func) {
+    if (observeThrottling && !IS_TESTING) {
+      long currentTimestamp = System.currentTimeMillis();
+      // Throttle to one report per interval.
+      if (currentTimestamp - generalTimestamp < GENERAL_REPORT_INTERVAL) {
+        return;
+      }
+      generalTimestamp = currentTimestamp;
+    }
+    func.accept(FlutterInitializer.getAnalytics());
+  }
+
+  private void maybeReportErrorCounts() {
+    long currentTimestamp = System.currentTimeMillis();
+    // Send accumulated error counts once every defined interval, plus when the project is closed.
+    if (errorsTimestamp == 0L || currentTimestamp - errorsTimestamp > ERROR_REPORT_INTERVAL || IS_TESTING) {
+      errorsTimestamp = currentTimestamp;
+      Analytics analytics = FlutterInitializer.getAnalytics();
+      LOG.debug(DAS_STATUS_EVENT_TYPE + " " + errorCount + " " + warningCount + " " + hintCount + " " + lintCount);
+      if (errorCount > 0) {
+        analytics.sendEventMetric(DAS_STATUS_EVENT_TYPE, ERRORS, errorCount); // test: serverStatus()
+      }
+      if (warningCount > 0) {
+        analytics.sendEventMetric(DAS_STATUS_EVENT_TYPE, WARNINGS, warningCount); // test: serverStatus()
+      }
+      if (hintCount > 0) {
+        analytics.sendEventMetric(DAS_STATUS_EVENT_TYPE, HINTS, hintCount); // test: serverStatus()
+      }
+      if (lintCount > 0) {
+        analytics.sendEventMetric(DAS_STATUS_EVENT_TYPE, LINTS, lintCount); // test: serverStatus()
+      }
+      errorCount = warningCount = hintCount = lintCount = 0;
```

Comment:
Why do we zero out the counts? This seems wrong. I would expect these #s should match the # of errors reported in the analysis server window. As is, I'm not clear how I would interpret these #s.

Reply:
They are accumulated while analysis is active, then sent when analysis is complete. They need to be zeroed so the accumulated values are accurate.